promutils/labeled/counter.go

```diff
@@ -59,8 +68,15 @@ func NewCounter(name, description string, scope promutils.Scope, opts ...MetricO
 		if _, emitUnlabeledMetric := opt.(EmitUnlabeledMetricOption); emitUnlabeledMetric {
 			c.Counter = scope.MustNewCounter(GetUnlabeledMetricName(name), description)
 		} else if additionalLabels, casted := opt.(AdditionalLabelsOption); casted {
-			c.CounterVec = scope.MustNewCounterVec(name, description, append(metricStringKeys, additionalLabels.Labels...)...)
-			c.additionalLabels = contextutils.MetricKeysFromStrings(additionalLabels.Labels)
+			var labels []string
+			for _, label := range additionalLabels.Labels {
+				if contains(metricStringKeys, label) == false {
+					labels = append(labels, label)
+				}
+			}
```

Comment:
Nit:
```suggestion
			labels := make([]string, 0, len(additionalLabels.Labels)
			metricKeysSet := sets.NewString(metricStringKeys...)
			for _, label := range additionalLabels.Labels {
				if !metricKeysSet.Contains(label) {
					labels = append(labels, label)
				}
			}
```

Reply:
And then we can delete the contains function

Reply:
updated it, thank you @EngHabu